python/pyspark/tests/test_worker.py

```diff
@@ -150,6 +151,20 @@ def test_with_different_versions_of_python(self):
         finally:
             self.sc.pythonVer = version
 
+    def test_python_exception_non_hanging(self):
+        # SPARK-21045: exceptions with no ascii encoding shall not hanging PySpark.
+        try:
+            def f():
+                raise Exception("exception with 中 and \xd6\xd0")
+
+            self.sc.parallelize([1]).map(lambda x: f()).count()
+        except Py4JJavaError as e:
+            if sys.version_info.major < 3:
+                # we have to use unicode here to avoid UnicodeDecodeError
+                self.assertRegexpMatches(unicode(e).encode("utf-8"), "exception with 中")
```

Comment:
do not we need to do `unicode = str` in test_worker.py?

Reply:
I ran dev/lint-python on my local machine, check passes, flake8 doesn't complain this one.  Not sure how it works. 

Let me push a new one for add `unicode = str` in test_worker.py?

Reply:
Are you sure your local flake8 runs with Python3?

Reply:
I just copied the change in test_worker.py and run dev/lint-python:

```
dev/lint-python 
starting python compilation test...
python compilation succeeded.

downloading pycodestyle from https://raw.githubusercontent.com/PyCQA/pycodestyle/2.4.0/pycodestyle.py...
starting pycodestyle test...
pycodestyle checks passed.

starting flake8 test...
flake8 checks failed:
./python/pyspark/sql/utils.py:31:60: F821 undefined name 'unicode'
        if sys.version_info.major < 3 and isinstance(desc, unicode):
                                                           ^
1     F821 undefined name 'unicode'
1
1
```

Reply:
Yes it fails only at utils.py not test_worker.py

Reply:
Oh, interesting.

For test_worker.py, it reports:

```
python/pyspark/tests/test_worker.py:162:73: F405 'unicode' may be undefined, or defined from star imports: pyspark.tests.test_worker
```
That is just warning. So it is fine.